### ISSUE TITLE
Use gas estimation

### ISSUE THREAD

**Issue by sanderpick** [recallnet/rust-recall#136]
Blocked by https://github.com/consensus-shipyard/ipc/issues/1183 (https://github.com/consensus-shipyard/ipc/pull/1185). Without this, the estimates aren't very good.

https://github.com/hokunet/rust-hoku/pull/129 enforces some client-side limits on fee cap and premium, which can be modified once this is done.

**Comment by sanderpick** [comment]
@avichalp is this done?

**Comment by sanderpick** [comment]
cool, sounds like a new issue would be useful. I created this one https://github.com/recallnet/rust-recall/issues/224
